tests/ProductCarousel.test.jsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * @jest-environment jsdom
+ */
+
+// Product Carousel Tests ==============================================
+
+import React from 'react';
+import {
+  render, cleanup, waitFor, fireEvent, screen,
+} from '@testing-library/react';
+import '@testing-library/jest-dom';
+import 'regenerator-runtime/runtime';
+import { Provider } from 'react-redux';
+import axios from 'axios';
+
+import ProductCarousel from '../client/src/components/Related/ProductCarousel';
+import Modal from '../client/src/components/Related/Modal';
+import RelatedItems from '../client/src/components/Related/RelatedItems';
+import store from '../client/src/store/store';
+
+beforeAll(() => {
+  axios.get.mockImplementation((url) => {
+    switch (url) {
+      case '/products/1':
+        return Promise.resolve(mockProductData);
+      case '/products/1/related':
+        return Promise.resolve(mockRelatedIds);
+      case '/products/48421/styles':
+        return Promise.resolve(mockStyleData);
+      default:
+        return Promise.reject(new Error('Error - this test is not working'));
+    }
+  });
+});
+
+afterEach(cleanup);
+
+jest.mock('axios');
+jest.mock('../client/src/components/Related/card.css', () => () => (<div>Carousel Card Style Placeholder</div>));
+jest.mock('../client/src/components/Related/carousel.css', () => () => (<div>Carousel Style Placeholder</div>));
+jest.mock('../client/src/components/Related/modal.css', () => () => (<div>PlModal Style Placeholder</div>));
+
```

Comment:
Add a jest.mock for Modal.jsx, I think circleCI is throwing an error because of it even though your tests look good

Reply:
jest.mock test added for Modal